frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -580,67 +582,92 @@
+                                }
                             }
-                        }
-                        else if (duration._data.seconds != 0) {
-                            var second = duration.asSeconds();
-                            vm.leaderboard[i].submission__submitted_at = second;
-                            if (second.toFixed(0)==1) {
-                                vm.leaderboard[i].timeSpan = 'second';
-                            } else {
-                                vm.leaderboard[i].timeSpan = 'seconds';
+                            else if (duration._data.days !=0) {
+                                var days = duration.asDays();
+                                vm.leaderboard[i].submission__submitted_at = days;
+                                if (days.toFixed(0)==1) {
+                                    vm.leaderboard[i].timeSpan = 'day';
+                                } else {
+                                    vm.leaderboard[i].timeSpan = 'days';
+                                }
+                            }
+                            else if (duration._data.hours !=0) {
+                                var hours = duration.asHours();
+                                vm.leaderboard[i].submission__submitted_at = hours;
+                                if (hours.toFixed(0)==1) {
+                                    vm.leaderboard[i].timeSpan = 'hour';
+                                } else {
+                                    vm.leaderboard[i].timeSpan = 'hours';
+                                }                        
+                            } 
+                            else if (duration._data.minutes !=0) {
+                                var minutes = duration.asMinutes();
+                                vm.leaderboard[i].submission__submitted_at = minutes;
+                                if (minutes.toFixed(0)==1) {
+                                    vm.leaderboard[i].timeSpan = 'minute';
+                                } else {
+                                    vm.leaderboard[i].timeSpan = 'minutes';
+                                }
+                            }
+                            else if (duration._data.seconds != 0) {
+                                var second = duration.asSeconds();
+                                vm.leaderboard[i].submission__submitted_at = second;
+                                if (second.toFixed(0)==1) {
+                                    vm.leaderboard[i].timeSpan = 'second';
+                                } else {
+                                    vm.leaderboard[i].timeSpan = 'seconds';
+                                }
                             }
                         }
-                    }
+                        if (vm.leaderboard.length == privateEntriesCount) {
+                            vm.allLeaderboardEntriesPrivate = false;
+                        }
+                    }, 2000);
```

Comment:
Why are you wrapping it in a timeout code?

Reply:
to get the `phaseSplits`, without timeout `phaseSplits` array output as empty

Reply:
Why would `phaseSplits` be an empty array? Did you look into the reason?